### PR TITLE
WT-7075 Prevent MongoDB yielding in the LSWA workload

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
@sulabhM 
I'm just putting this here to make it visible to you.

We're not seeing much lookaside usage yet but the transaction length looks good and the yielding has been solved. Reducing the document count and increasing the size hasn't helped as we'd anticipated.

### CODE REVIEW

File: src/cast_core/src/CollectionScanner.cpp [hunk 1]
```diff
@@ -107,7 +107,7 @@ struct CollectionScanner::PhaseConfig {
         } else if (scanTypeString == "snapshot") {
             transactionOptions = mongocxx::options::transaction{};
             auto readConcern = mongocxx::read_concern{};
-            readConcern.acknowledge_level(mongocxx::read_concern::level::k_majority);
+            readConcern.acknowledge_level(mongocxx::read_concern::level::k_snapshot);
```
Comment (tetsuo-cpp) [inline src/cast_core/src/CollectionScanner.cpp:110]: Necessary to prevent yielding on the `getMore`.

File: src/phases/scale/LargeScalePhases.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ GlobalDefaults:
   Random30String: &rand_30b_string {^RandomString: {length: 30}}
   Random130String: &rand_100b_string {^RandomString: {length: 100}}
 
-  Duration: &Duration 12 hours
+  Duration: &Duration 2 hours
```
Comment (tetsuo-cpp) [inline src/phases/scale/LargeScalePhases.yml:17]: For debugging.

File: src/phases/scale/LargeScalePhases.yml [hunk 3]
```diff
@@ -65,7 +65,7 @@ LongLivedCreatorCmd:
     x6: *rand_4b_int
     x7: *rand_4b_int
     x8: *rand_4b_int
-    s0: {^RandomString: {length: {^RandomInt: {min: 20, max: 80}}}}
+    s0: {^RandomString: {length: {^RandomInt: {min: 2000, max: 8000}}}}
```
Comment (tetsuo-cpp) [inline src/phases/scale/LargeScalePhases.yml:68]: Creating larger docs.

File: src/phases/scale/LargeScalePhases.yml [hunk 4]
```diff
@@ -237,9 +237,9 @@ SnapshotScanner5GigabytesCmd:
   GlobalRate: 1 per 5 minutes
 
 # Leave out the ScanDuration since we'll be supplying the ScanRateMegaBytes option.
-SnapshotScanner5GigabytesFixedRateCmd:
+SnapshotScanner10GigabytesFixedRateCmd:
   Duration: *Duration
-  ScanSizeBytes: 500000000
+  ScanSizeBytes: 10000000000
```
Comment (tetsuo-cpp) [inline src/phases/scale/LargeScalePhases.yml:242]: I wanted to increase the scan rate (which necessitates increasing the size if we want the same duration).

**Comment by tetsuo-cpp** [comment]
I'm running with this script to prevent yielding:

```
#!/bin/bash

killall -9 mongod; rm -rf rplset/rs0/mongod.log; rm -rf rplset/rs0/db/; mkdir rplset/rs0/db; bin/mongod -f mongod.conf; sleep 1; bin/mongo --eval "rs.initiate()"; bin/mongo --eval "db.adminCommand({setParameter: 1, internalQueryExecYieldIterations: 2147483647})"; bin/mongo --eval "db.adminCommand({setParameter: 1, internalQueryExecYieldPeriodMS: 2147483647})"; sleep 2
```

File: src/cast_core/src/CollectionScanner.cpp (hunk 1 above)
Comment (rtimmons) [reply src/cast_core/src/CollectionScanner.cpp:110]: Consider adding this comment to the source for the next maintainer 😄

**Review by rtimmons — APPROVED** [review]
LGTM assuming a successful patch-build.

(Hopefully you weren't waiting on me to merge - sorry if you were. In general getting LGTM from STM on individual actors/workload yamls isn't necessary, but we're happy to be another set of eyes if there's any uncertainty.)

**Comment by tetsuo-cpp** [comment]
> (Hopefully you weren't waiting on me to merge - sorry if you were. In general getting LGTM from STM on individual actors/workload yamls isn't necessary, but we're happy to be another set of eyes if there's any uncertainty.)

No worries! We're in the middle of debugging the workload's interaction with Mongo so that's why it's been sitting here.

**Comment by tetsuo-cpp** [comment]
@rtimmons The failed job is also failing in Waterfall so I'm going to go ahead and merge this.

**Pull request merged**